### PR TITLE
[CorrDiff]: Only pass interface_kwargs into ResLoss constructor, but empty for EDMLossSR

### PR DESCRIPTION
**Pull request by DavidLandup0**
# Modulus Pull Request

## Description
The PR sets the`interface_kwargs` to `{}` if the loss type is set to `modulus.metrics.diffusion.EDMLossSR` in CorrDiff's `training_loop.py`, and uses the existing kwargs for `modulus.metrics.diffusion.ResLoss`. The kwargs being passed into it are:

```
interface_kwargs = dict(
            regression_net=net_reg,
            img_shape_x=img_shape_x,
            img_shape_y=img_shape_y,
            patch_shape_x=patch_shape_x,
            patch_shape_y=patch_shape_y,
            patch_num=patch_num,
            hr_mean_conditioning=hr_mean_conditioning,
        )
```

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus/issues/537) is linked to this pull request.

/cc @mnabian

### CODE REVIEW

File: examples/generative/corrdiff/training/training_loop.py [hunk 1]
```diff
@@ -177,15 +177,19 @@ def training_loop(
...
-            regression_net=net_reg,
-            img_shape_x=img_shape_x,
-            img_shape_y=img_shape_y,
-            patch_shape_x=patch_shape_x,
-            patch_shape_y=patch_shape_y,
-            patch_num=patch_num,
-            hr_mean_conditioning=hr_mean_conditioning,
-        )
+
+        if loss_kwargs["class_name"] == "modulus.metrics.diffusion.ResLoss":
```
Comment (Alexey-Kamenev) [inline examples/generative/corrdiff/training/training_loop.py:181]: Just checking that `class_name` will always be in `loss_kwargs` (so you don't get `KeyError`)?

**Review by Alexey-Kamenev — APPROVED** [review]

File: examples/generative/corrdiff/training/training_loop.py (hunk 1 above)
Comment (mnabian) [reply examples/generative/corrdiff/training/training_loop.py:181]: Yes, it will be always in the loss kwargs.

**Pull request merged**